### PR TITLE
.NET 3.4.1 Release notes

### PR DESCRIPTION
**Pull request by jeffrymorris**

### CODE REVIEW

File: modules/project-docs/pages/sdk-release-notes.adoc [hunk 1]
```diff
@@ -13,6 +13,30 @@ These pages cover the 3._x_ versions of the Couchbase .NET SDK.
 For release notes, download links, and installation methods for 2.7 and earlier releases of the Couchbase .NET Client,  please see the xref:2.7@dotnet-sdk::sdk-release-notes.adoc[2.x .NET Release Notes & Download Archive].
 
 The full installation instructions that were previously on this page can now be found xref:sdk-full-installation.adoc[here].
+[#version-3-4-1]
+== Version 3.4.1 (09 December 2022)
+
+Version 3.4.1 is the second release of the 3.4 series.
```
Comment (dnault) [inline modules/project-docs/pages/sdk-release-notes.adoc:19]: NIT: This is information-free :-) Omit it if you want.

File: modules/project-docs/pages/sdk-release-notes.adoc [hunk 2]
```diff
@@ -13,6 +13,30 @@ These pages cover the 3._x_ versions of the Couchbase .NET SDK.
...
+
+=== Special Note
+
+* During a rebalance upgrade from 6.x (or any earlier version) to 7.x, in mixed mode (where you are communicating with Couchbase Server whilst some but not all nodes are upgraded), there is a known issue where data may be written to the wrong location.  
+The solution is to either upgrade to `3.2.9` or greater, or to pause application processing so there are no writes until you have upgraded all nodes.  
+If you encounter a similar situation during migration and need help with mitigation, please contact our support team.
+
+=== Fixed Issues
+
+* https://issues.couchbase.com/browse/NCBC-3204[NCBC-3204]: CombinationTest failure: Test_GetAndLockAsync_Locked
```
Comment (dnault) [inline modules/project-docs/pages/sdk-release-notes.adoc:29]: NIT: If the fix was to change/disable the test, is it release note-worthy?

**Review by dnault — APPROVED** [review]
LGTM.

I can't remember whose idea this was (Richard? Hakim? Maria?), but over in Java-land we've been trying to phrase the release notes as positives instead of negatives. In other words, describing the improvement instead of the flaw.

For example, instead of "Search: Min function throws an exception if the argument is > 0", we could say:

> When using a DisjunctionQuery in a full-text search, it's now possible to specify the minimum number of sub-queries that must be satisfied. (Previously, `DisjunctionQuery.Min()` would always throw `ArgumentOutOfRangeException`.)

File: modules/project-docs/pages/sdk-release-notes.adoc (hunk 1 above)
Comment (RichardSmedley) [reply modules/project-docs/pages/sdk-release-notes.adoc:19]: Thanks David - this bit is as much visual pause as orientation, for those skimming quickly.

**Comment by RichardSmedley** [comment]
> LGTM.
> 
> I can't remember whose idea this was (Richard? Hakim? Maria?), but over in Java-land we've been trying to phrase the release notes as positives instead of negatives. In other words, describing the improvement instead of the flaw.

Thanks David - we've asked for this before, but the .NET team doesn't have the resources at present.

One workaround we've discussed with @raycardillo is a compulsory `release notes` field in JIRA, so that the text can be filled out when closing an issue - whilst the knowledge is fresh in the engineer's mind.

**Pull request merged**